tests/testthat/test-S3rspec.R

```diff
@@ -137,10 +137,12 @@ test_that("merge.rspec", {
   expect_s3_class(sicalis_merged, "rspec")
   expect_mapequal(sicalis_TC, sicalis_merged)
 
-  expect_error(
-    merge(sicalis_T, as.data.frame(sicalis_C)),
-    "invalid rspec objects"
-  )
+  # This now works as it should (by combining both into a single rspec object) 
+  # with the rewritten function. It should, shouldn't it?
```

Comment:
Yes, it should run without error but I think it'll give an unexpected output. From what I understand, the second argument will be silently ignored.
This example is a good demonstration of why this change might bring more trouble than it solves.